wfl/Controllers/RestaurantsController.cs

```diff
@@ -23,33 +23,43 @@ public RestaurantsController(IDBManager dbManager)
             {
                 return BadRequest();
             }
-            Restaurant r = await DBManager.InsertRestaurantAsync(restaurant);
+            Restaurant r = new Restaurant()
+            {
+                Name = restaurant.Name,
+                MenuLink = restaurant.MenuLink
+            };
+            r = await DBManager.InsertEntityAsync(r);
             return Created(r.Id.ToString(), r);
         }
 
         [Route("{id}")]
         [HttpPut]
         public async Task<IHttpActionResult> UpdateAsync([FromUri] string id, [FromBody] RestaurantDto restaurant)
         {
-            if (string.IsNullOrEmpty(id) || await DBManager.GetRestaurantWithIdAsync(id) == null || string.IsNullOrEmpty(restaurant?.Name) || (restaurant.MenuLink != null && !Uri.IsWellFormedUriString(restaurant.MenuLink.ToString(), UriKind.Absolute)))
+            if (string.IsNullOrEmpty(id) || await DBManager.GetEntityByIdAsync<Restaurant>(id) == null || string.IsNullOrEmpty(restaurant?.Name) || (restaurant.MenuLink != null && !Uri.IsWellFormedUriString(restaurant.MenuLink.ToString(), UriKind.Absolute)))
             {
                 return BadRequest();
             }
-            return Ok(await DBManager.UpdateRestaurantAsync(id, restaurant));
+            Restaurant r = new Restaurant()
```

Comment:
I was under the impression that a PUT should replace the whole entity. I'll add these changes.